File: leetcode/c++/14.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.size() == 0) {
            return "";
        }
        if (strs.size() == 1) {
            return strs[0];
        }
        int n = strs[0].size();
        for (int i = 1; i < strs.size(); i++) {
            int m = strs[i].size();
            if (m < n) {
                n = m;
            }
        }
        if (n == 0) {
            return "";
        }
        char *chars = new char[n + 1];
        int count = 0;
        while (count < n) {
            char ch = strs[0][count];
            bool match = true;
            for (int i = 1; i < strs.size(); i++) {
                if (ch != strs[i][count]) {
                    match = false;
                    break;
                }
            }
            if (!match) {
                break;
            }
            chars[count++] = ch;
        }
        chars[count] = '\0';
        string res(chars, chars + count);
        delete[] chars;
        return res;
    }
};

File: leetcode/python/14.py
#!/usr/bin/env python

__author__ = "yangyanzhan"
__email__ = "yangyanzhan@gmail.com"
__url__ = "https://github.com/yangyanzhan/code-camp"
__blog__ = "https://yangyanzhan.github.io"

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 0:
            return ""
        if len(strs) == 1:
            return strs[0]
        n = len(strs[0])
        for i in range(1, len(strs)):
            m = len(strs[i])
            n = min(n, m)
        if n == 0:
            return ""
        res = ""
        for i in range(n):
            ch = strs[0][i]
            for j in range(1, len(strs)):
                if strs[j][i] != ch:
                    return res
            res += ch
        return res
